File: bfinter.java
import java.util.Scanner;
import java.util.Arrays;

public class bfinter {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String program = sc.nextLine();
        int memorySize = 65536;
        int[] arr = new int[memorySize];
        Arrays.fill(arr, 0);
        int pointer = 0;
        int bra = 0;
        int i = 0;
        while (i < program.length()) {
            if (program.charAt(i) == '+') {
                if (0 <= pointer && pointer < memorySize) {
                    arr[pointer]++;
                } else {
                    System.out.println("Out Of Array Bounds At:" + i);
                    return;
                }
                i++;
            } else if (program.charAt(i) == '-') {
                if (0 <= pointer && pointer < memorySize) {
                    arr[pointer]--;
                } else {
                    System.out.println("Out Of Array Bounds At:" + i);
                    return;
                }
                i++;
            } else if (program.charAt(i) == '<') {
                pointer--;
                i++;
            } else if (program.charAt(i) == '>') {
                pointer++;
                i++;
            } else if (program.charAt(i) == ',') {
                int inp = sc.nextInt();
                arr[pointer] = inp;
                i++;
            } else if (program.charAt(i) == '.') {
                System.out.print((char) arr[pointer]);
                i++;
            } else if (program.charAt(i) == '[') {
                if (arr[pointer] == 0) {
                    bra = 1;
                    while (bra > 0) {
                        i++;
                        if (i < program.length() && program.charAt(i) == '[') {
                            bra++;
                        } else if (i < program.length() && program.charAt(i) == ']') {
                            bra--;
                        } else if (i >= program.length()) {
                            System.out.println("EOF Reached");
                            return;
                        } else {
                        }
                    }
                } else {
                    i++;
                }
            } else if (program.charAt(i) == ']') {
                if (arr[pointer] != 0) {
                    bra = 1;
                    while (bra > 0) {
                        i--;
                        if (i >= 0 && program.charAt(i) == '[') {
                            bra--;
                        } else if (i >= 0 && program.charAt(i) == ']') {
                            bra++;
                        } else if (i < 0) {
                            System.out.println("Out of Program");
                            return;
                        } else {
                            //Other charcters.
                        }
                    }
                } else {
                    i++;
                }
            }
        }
    }
}

File: bfinter.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
	string program;
	int memorysize = 65536;
	vector<int> arr(memorysize, 0);
	int pointer = 0;
	int bra = 0;
	int i = 0;
	cin >> program;
	while (i < program.length())
	{
		if (program[i] == '+')
		{
			if (0 <= pointer && pointer < memorysize)
			{
				arr[pointer]++;
			}
			else
			{
				cout << "Out Of Array Bounds At:" + i << endl;
				return 0;
			}
			i++;
		}
		else if (program[i] == '-')
		{
			if (0 <= pointer && pointer < memorysize)
			{
				arr[pointer]--;
			}
			else
			{
				cout << "Out Of Array Bounds At:" + i << endl;
				return 0;
			}
			i++;
		}
		else if (program[i] == '<')
		{
			pointer--;
			i++;
		}
		else if (program[i] == '>')
		{
			pointer++;
			i++;
		}
		else if (program[i] == '[')
		{
			if (arr[pointer] == 0)
			{
				bra = 1;
				while (bra > 0)
				{
					i++;
					if (i < program.length() && program[i] == '[')
					{
						bra++;
					}
					else if (i < program.length() && program[i] == ']')
					{
						bra--;
					}
					else if (i >= program.length())
					{
						cout << "EOF Reached" << endl;
						return 0;
					}
					else
					{
					}
				}
			}
			else
			{
				i++;
			}
		}
		else if (program[i] == ']')
		{
			if (arr[pointer] != 0)
			{
				bra = 1;
				while (bra > 0)
				{
					i--;
					if (i >= 0 && program[i] == '[')
					{
						bra--;
					}
					else if (i >= 0 && program[i] == ']')
					{
						bra++;
					}
					else if (i < 0)
					{
						cout << "Out of Program" << endl;
						return 0;
					}
					else
					{
					}
				}
			}
			else
			{
				i++;
			}
		}
		else if (program[i] == '.')
		{
			cout << ((char)arr[pointer]);
			i++;
		}
		else if (program[i] == ',')
		{
			int inp;
			cin >> inp;
			arr[pointer] = inp;
			i++;
		}
		else
		{
			i++;
		}
	}
	return 0;
}
